### PR TITLE
Create findings report

### PR DESCRIPTION
**Pull request by madelondohmen**
### Changes
This PR adds a Findings Report to the Generate Report flow, to get rid of the old LaTeX Findings Report, which was located in a different file than the other reports.

The structure looks like the other reports. It replaces the old findings report, which was located in the "partials" folder. This report is located in the "reports" folder, just like all the other reports. 

Besided this, changes have been made to some buttons:
- The "Download PDF" button on the findings page has been deleted*
- The "Generate Report" button on the finding details page and on the object details page now redirects to the second step of the "Setup" step in the Generate Report flow.

* Removing this button is temporary until the Aggregate Findings Report is created, but it makes sure that old Findings Report (in LaTeX) isn't used anymore.

### Important information
This pull request does not fully resolve issue #2278. To do this, an Aggregate Findings Report needs to be created first. However, implementing this feature poses a challenge as it is currently only possible to generate Z-CERT-specific aggregate reports. The addition of this functionality is necessary before removing any old files can take place. We'll need a clean-up, but before that we need the new aggregate report flow to work completely.

### Issue link
#2278 

### Demo
Findings report:
https://github.com/minvws/nl-kat-coordination/assets/99282220/82820e55-3d96-444f-b5d5-90e8adbb73f4

Deleted "Download PDF" button:
afbeelding

"Generate Report" now links to the Generate Findings Report for the selected OOI
https://github.com/minvws/nl-kat-coordination/assets/99282220/6bc841a2-8703-43a3-b844-f96f4b812289

---

### Code Checklist
- [x] All the commits in this PR are properly PGP-signed and verified.
- [x] This PR only contains functionality relevant to the issue; tickets have been created for newly discovered issues.
- [x] I have written unit tests for the changes or fixes I made.
- [ ] For any non-trivial functionality, I have added integration and/or end-to-end tests.
- [x] I have performed a self-review of my code and refactored it to the best of my abilities.

### Communication
- [x] I have informed others of any required `.env` changes files if required and changed the `.env-dist` accordingly.
- [x] I have made corresponding changes to the documentation, if necessary.
- [x] I have included comments in the code to elaborate on what is not self-evident from the code itself, including references to issues and discussions online, or implicit behavior of an interface.

---
## Checklist for code reviewers:
Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---
## Checklist for QA:
Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.

### CODE REVIEW

**Comment by noamblitz** [comment]
Im curious to know who uses the findings -> download PDF button. Since this functionality will be added back later, we might want to give the possibility for people to keep using it.

**Comment by underdarknl** [comment]
> Im curious to know who uses the findings -> download PDF button. Since this functionality will be added back later, we might want to give the possibility for people to keep using it.

Shoudn't the PDF and Json downloads for this report follow from the same route that all other reports follow?

**Comment by noamblitz** [comment]
Yes! For sure! It's just that the "old" Findings report of an organization will be replaced by the new Aggregate Findings Report. Which hasn't been built yet (blocked by new report flow). So if we remove the old Organization Findings Report, people that are using that will not have an alternative for maybe a release.

**Comment by stephanie0x00** [comment]
## Checklist for QA:
- [x] I have checked out this branch, and successfully ran a fresh `make reset`.
- [x] I confirmed that there are no unintended functional regressions in this branch:
    - [x] I have managed to pass the onboarding flow
    - [x] Objects and Findings are created properly
    - [x] Tasks are created and completed properly
- [x] I confirmed that the PR's advertised `feature` or `hotfix` works as intended.

### What works:
Findings report can be generated and contains all findings as shown on the Findings page. 

### What doesn't work:
- Findings aren't currently sorted from highest to lowest severity rating. 
- Required and suggested plugins aren't implemented yet. These are added to the list in #2278 

Unsure if we want this in this report or the aggregate report, however there is currently no summary of findings in this report. An end user would like see an overview of the number of findings per severity rating. 

### Bug or feature?:
n/a

**Comment by madelondohmen** [comment]
> Im curious to know who uses the findings -> download PDF button. Since this functionality will be added back later, we might want to give the possibility for people to keep using it.

Agreed. So the choice is to either leave the button (which generates a LaTeX-based Findings Report PDF) in place or to remove it completely until the new flow is built. It depends on how many users actually use this button and how quickly we want to get rid of the LaTeX code.

**Comment by noamblitz** [comment]
Agree with @stephanie0x00 that this should include the summary, we could then reuse the same template in the aggregate. New ticket?

**Comment by madelondohmen** [comment]
> Agree with @stephanie0x00 that this should include the summary, we could then reuse the same template in the aggregate. New ticket?

I've added the summary. No new ticket is needed.

File: rocky/reports/templates/partials/report_severity_totals_table.html
Comment (Donnype) [inline rocky/reports/templates/partials/report_severity_totals_table.html:10]: Not sure what we use in other tables, but perhaps because I was quickly wondering what "Unique" means we could instead say "Unique/Total Finding Types" or perhaps "Finding Types" even?

Comment (Donnype) [inline rocky/reports/templates/partials/report_severity_totals_table.html:10]: Not sure what we use in other tables, but perhaps because I was quickly wondering what "Unique" means we could instead say "Unique/Total Finding Types" or perhaps "Finding Types" even?

Comment (Donnype) [inline rocky/reports/templates/partials/report_severity_totals_table.html:10]: Not sure what we use in other tables, but perhaps because I was quickly wondering what "Unique" means we could instead say "Unique/Total Finding Types" or perhaps "Finding Types" even?

Comment (madelondohmen) [reply rocky/reports/templates/partials/report_severity_totals_table.html:10]: If that makes the table (and the code) more clear, I'll change it (also in the report.py). 

For continuity, I'd then go for "Finding types" and "Occurrences" (without the "Total"). What do you think?

Comment (Donnype) [reply rocky/reports/templates/partials/report_severity_totals_table.html:10]: I think that would indeed make sense!

**Comment by stephanie0x00** [comment]
## Checklist for QA:
- [x] I have checked out this branch, and successfully ran a fresh `make reset`.
- [x] I confirmed that there are no unintended functional regressions in this branch:
    - [x] I have managed to pass the onboarding flow
    - [x] Objects and Findings are created properly
    - [x] Tasks are created and completed properly
- [x] I confirmed that the PR's advertised `feature` or `hotfix` works as intended.

### What works:
Report is nicely generated as expected. Discussion point below. If we decide not to make any chances it is ready for merge after Review is also okay with this. 

### What doesn't work:
n/a

### Bug or feature?:
I'm unsure if this is something we want to fix in this version, or want to have this fixed in a future ticket. When selecting the objects I can basically select all objects for my domain. When generating the findings report, this will result in the same findings shown 4 times (I selected 4 objects as shown below). 

The screenshot below shows that 4 objects are selected. 
findingsreport

The screenshot below shows the 'start' of one of the findings reports for one of the objects. The report for the other objects seems to be the same. So the question is as follow:
- Do we want to show all findings for all objects, when they are 100% similar? 
- Does the current findings report lack a filter (e.g. that we only show the findings for the Hostname, instead of also DNSZone, URL, HostnameHTTPURL, etc.).
- Is this expected behaviour for the current report transition? 
- Other?  

findingsreport-4times

File: rocky/reports/report_types/findings_report/report.py
Comment (underdarknl) [inline rocky/reports/report_types/findings_report/report.py:39]: could we make the tree_depth user-selected? And if so, also include the tree depth used in the resulting report.

Comment (underdarknl) [reply rocky/reports/report_types/findings_report/report.py:39]: Also, if possible, can we have Octopoes/xtdb do the culling of this tree based on the OOI type instead of the report? Im not sure the current api's support this, but it feels like a reasonable feature to want.

File: rocky/rocky/templates/partials/ooi_detail_toolbar.html
Comment (underdarknl) [inline rocky/rocky/templates/partials/ooi_detail_toolbar.html:6]: Lets update this link to say 'Generate findings report'

File: rocky/reports/report_types/findings_report/report.py
Comment (madelondohmen) [reply rocky/reports/report_types/findings_report/report.py:39]: > could we make the tree_depth user-selected? And if so, also include the tree depth used in the resulting report.

Yes, we could. But to do this, I think we first need to create the configuration page, which is currently still in design.

File: rocky/reports/templates/partials/report_severity_totals.html
Comment (dekkers) [inline rocky/reports/templates/partials/report_severity_totals.html:12]: I think that including `:` in the translated string makes it easier for translator to know what the context is.

```suggestion
                <span>{% translate "See:" %} <a href="https://nvd.nist.gov/vuln-metrics/cvss"
```

(The translation files are need to be updated when the suggestion is committed)

File: rocky/reports/report_types/findings_report/report.py
Comment (dekkers) [reply rocky/reports/report_types/findings_report/report.py:39]: The goal was to reimplement the current keiko findings report and as far as I can see this has been done. If we want to make changed how the report is generated I think it would be better to create new tickets for that.

I am also not sure if asking the user for the tree depth would be good UX, because the user probably doesn't  know what to specify.

File: rocky/reports/report_types/findings_report/report.py
Comment (dekkers) [inline rocky/reports/report_types/findings_report/report.py:61]: Finding types are not guaranteed to have a risk score (a new CVE might not have scoring information yet).

```suggestion
        finding_types = sorted(finding_types.values(), key=lambda x: x["finding_type"].risk_score or 0, reverse=True)
```

**Review by dekkers — COMMENTED** [review]
We are currently writing unit tests for reports that don't have them yet, so I think it would be good to also add unit tests for this report.

**Comment by madelondohmen** [comment]
> We are currently writing unit tests for reports that don't have them yet, so I think it would be good to also add unit tests for this report.

Unit tests have been added now.

**Comment by madelondohmen** [comment]
Changes have been made to the design, conform the new design of the Findings Report. 

https://www.figma.com/file/QRL8085Z827bNdO0bIDdFL/KAT-%7C-Design-%7C-For-review?type=design&node-id=3287-21907&mode=design

Changes may need to be made to the HTML, but this will be done later (or in a new PR).

File: rocky/reports/report_types/findings_report/report.py
Comment (underdarknl) [reply rocky/reports/report_types/findings_report/report.py:44]: i was under the impression the original report also had a configurable tree depth, if not I agree with creating a new ticket for this and fixing this once report configs land.

Comment (dekkers) [reply rocky/reports/report_types/findings_report/report.py:44]: Hmm you are right, I thought this always used the default, but on the report page there is a depth parameter in the form that is hidden by default. We don't really have a way to specify these things in the current reporting flow however.

Maybe we should revert the change to "generate report" button on the ooi page for now so that we can merge the rest of the code?

**Review by underdarknl — APPROVED** [review]

**Pull request merged**
